### PR TITLE
Fix bootstrap log publish

### PR DESCRIPTION
**Pull request by jaredpar**
PR #71265 tried to unify machine cleanup on CI. It did not set the `$prepareMachine` value for our bootstrap builds. That meant that `VBCSCompiler` kept running after the build as `ExitWithExitCode` did not kill processes and it held onto log files that blocked publishing. That results in us being able to investigate bootstrap failures (see #71591).

The log file is opened with `FileShare.ReadWrite` but apparently that is not enough for publish artifacts to work

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure PTAL

**Review by RikkiGibson — APPROVED** [review]
Did we open a PR based on this which breaks bootstrap, in order to show that the change causes the logs to be published? It's probably fine if we want to verify this after merge, just thought I would check.

**Review by Cosifne — APPROVED** [review]

**Comment by RikkiGibson** [comment]
Trying it out in #71591

**Comment by jaredpar** [comment]
The `main` branch is in lockdown but I verified with both @333fred and @Cosifne that they are comfortable with me merging this anyways as it impacts GitHub only and helps unblock other investigations.

**Pull request merged**
